scio-core/src/main/scala/com/spotify/scio/coders/KryoAtomicCoder.scala

```diff
@@ -23,15 +23,54 @@ import com.google.cloud.dataflow.sdk.coders.Coder.Context
 import com.google.cloud.dataflow.sdk.coders._
 import com.google.cloud.dataflow.sdk.util.VarInt
 import com.google.common.io.ByteStreams
+import com.google.common.reflect.ClassPath
 import com.google.protobuf.Message
 import com.twitter.chill._
 import com.twitter.chill.algebird.AlgebirdRegistrar
 import com.twitter.chill.protobuf.ProtobufSerializer
 import org.apache.avro.generic.GenericRecord
 import org.apache.avro.specific.SpecificRecordBase
+import org.slf4j.LoggerFactory
 
+import scala.collection.JavaConverters._
 import scala.collection.convert.Wrappers.JIterableWrapper
 
+@KryoRegistrar
+private class AlgebirdKryoRegistrar extends IKryoRegistrar {
+  override def apply(k: Kryo): Unit = new AlgebirdRegistrar()(k)
+}
+
+private object KryoRegistrarLoader {
+
+  private val logger = LoggerFactory.getLogger("KryoRegistrarLoader")
+
+  def load(k: Kryo): Unit = {
+    logger.info("Loading KryoRegistrars: " + registrars.mkString(", "))
+    registrars.foreach(_(k))
+  }
+
+  private val registrars: Seq[IKryoRegistrar] = {
+    logger.info("Initializing KryoRegistrars")
+    val classLoader = Thread.currentThread().getContextClassLoader
+    ClassPath.from(classLoader).getAllClasses.asScala.toSeq
+      .filter(_.getName.endsWith("KryoRegistrar"))
```

Comment:
The next `flatMap` can be very slow without this fillter, probably due to the cost of `clsInfo.load`. Thoughts?